lang/cs/Org.Apache.REEF.Common/Telemetry/MetricsData.cs

```diff
@@ -0,0 +1,240 @@
+        [JsonConstructor]
+        internal MetricsData(string serializedMetricsString)
+        {
+            var metrics = JsonConvert.DeserializeObject<IList<MetricData>>(serializedMetricsString, settings);
+            foreach (var m in metrics)
+            {
+                _metricsMap.Add(m.GetMetric().Name, m);
+            }
+        }
+
+        internal MetricsData(IMetrics metrics)
+        {
+            foreach (var me in metrics.GetMetrics())
+            {
+                _metricsMap.Add(me.GetMetric().Name, new MetricData(me.GetMetric()));
+            }
+        }
+
+        /// <summary>
+        /// Checks if the metric to be registered has a unique name. If the metric name has already been 
+        /// registered, metric is not entered into the registration and method returns false. On successful
+        /// registration, method returns true.
+        /// </summary>
+        /// <param name="metric">Metric to register.</param>
+        /// <returns>Indicates if the metric was registered.</returns>
+        public bool TryRegisterMetric(IMetric metric)
+        {
+            lock (_metricLock)
+            {
+                if (_metricsMap.ContainsKey(metric.Name))
+                {
+                    Logger.Log(Level.Warning, "The metric [{0}] already exists.", metric.Name);
+                    return false;
+                }
+                _metricsMap.Add(metric.Name, new MetricData(metric));
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Gets a metric given a name.
+        /// </summary>
+        /// <param name="name">Name of the metric.</param>
+        /// <param name="me">The metric object returned.</param>
+        /// <returns>Boolean indicating if a metric object was succesfully retrieved.</returns>
+        public bool TryGetValue(string name, out IMetric me)
+        {
+            lock (_metricLock)
+            {
+                if (!_metricsMap.TryGetValue(name, out MetricData md))
+                {
+                    me = null;
+                    return false;
+                }
+                me = md.GetMetric();
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Gets all the registered metrics.
+        /// </summary>
+        /// <returns>IEnumerable of MetricData.</returns>
+        public IEnumerable<MetricData> GetMetrics()
+        {
+            return _metricsMap.Values;
+        }
+
+        /// <summary>
+        /// Updates metrics given another <see cref="MetricsData"/> object.
+        /// For every metric in the new set, if it is registered then update the value,
+        /// if it is not then add it to the registration.
+        /// </summary>
+        /// <param name="metrics">New metric values to be updated.</param>
+        internal void Update(MetricsData metrics)
+        {
+            lock (_metricLock)
+            {
+                foreach (var metric in metrics.GetMetrics())
```

Comment:
Could we avoid the use of "var" when the data type is obvious? The reason is that expanding on the data type increases readability.